### ISSUE TITLE
Record every Training Run with MLflow, including the Checkpoint it produced

### ISSUE THREAD

**Issue by pergazuz** [pergazuz/LB-anythings#12]
## What to build

Every Training Run records its metrics, its hyperparameters and a copy of the Checkpoint it
produced, so that the Operator can see whether the Detector is improving and can go back to a
better Checkpoint.

Today a Training Run records `id`, `pid` and `started_at` and nothing else, and it writes into
`runs/<run name>` with `exist_ok=True`. So each run destroys the previous run's Checkpoint and
its results, and neither of the two questions that matter can be answered: *is the Detector
getting better as I label?* and *last week's was better, can I have it back?*

Ultralytics ships an MLflow callback and it is already enabled in this machine's ultralytics
settings; it no-ops only because the library is not installed. It logs every training argument
as a parameter, the losses and mAP50/mAP50-95 per epoch as metrics, and
`mlflow.log_artifact(trainer.best.parent)` — the whole weights directory, which is what fixes
the destroyed-Checkpoint half of the problem. So this slice is mostly wiring.

## Acceptance criteria

- [ ] `mlflow` is a dependency group of its own, in the default groups, so `uv sync` gives a
      working tracked setup and `uv sync --no-default-groups --group dev` (what CI runs) does
      not. No test requires mlflow to be installed.
- [ ] `LB_TRACKING_URI` selects where runs are recorded, defaulting to `<data dir>/mlflow`, and
      is resolved before it reaches ultralytics for the same reason `project` is (see #10).
- [ ] `LB_TRACKING_EXPERIMENT` names the experiment, defaulting to `lb-anythings`.
- [ ] `LB_TRACKING` turns it off without uninstalling anything, and does so **without writing
      to ultralytics' own machine-wide `settings.json`** — the mlflow callbacks are dropped
      from the model instance, not disabled globally.
- [ ] Each Training Run gets a distinct MLflow run name; every run showing as `active` in the
      UI would defeat the point.
- [ ] With mlflow absent the trainer behaves exactly as it does now, and says nothing about it.
- [ ] Verified against a real Training Run on the pipe Checkpoint: the run appears in
      `mlflow ui` with its metrics, its parameters, and `best.pt` as an artifact.
- [ ] The README documents the settings, how to open the UI, and the retention cost below.

## Retention

The pipe Checkpoint is 19 MB and the callback archives `best.pt` *and* `last.pt` per run, so a
Retrain Threshold of 25 costs roughly 40 MB per run — a couple of GB after fifty. Say so in the
README rather than letting it be discovered. A retention policy is out of scope here.

## Out of scope

- Anything the application ring knows and ultralytics does not: that is the follow-up ticket.
- Choosing a different tracker. W&B, Comet and ClearML are cloud-first and want an account;
  this is a single machine running local Label Studio on private footage. TensorBoard logs
  scalars but stores no artifacts, so it does not solve the destroyed-Checkpoint half.

**Comment by pergazuz** [comment]
Done in 4e066b3.

Tracking is its own default dependency group, so `uv sync` gives a recorded setup and CI's `--no-default-groups --group dev` does not. `LB_TRACKING`, `LB_TRACKING_URI` and `LB_TRACKING_EXPERIMENT` control it; the README documents the settings, the UI command and the retention cost.

## What the real runs found

**MLflow 3 has deprecated the plain-directory store and now raises instead of opening one.** The first implementation used a directory, and it recorded *nothing* while looking entirely successful: the trainer logged normally, the run finished, and `data/mlflow` stayed empty. Only the acceptance criterion that demanded a real run caught it. The default is now SQLite under the data directory; `LB_TRACKING_URI` takes any MLflow URI, so a shared server is `http://mlflow.local:5000`.

**The experiment must be created with an explicit `artifact_location`.** Otherwise MLflow archives into `./mlruns` relative to wherever the Training Run was spawned — the same trap as ultralytics' relative `project` in #10 — and creation is the only moment that location can be set.

Turning tracking off drops the mlflow callbacks from the model instance rather than flipping ultralytics' `mlflow` key, which lives in `%APPDATA%\Ultralytics\settings.json` and would change every other ultralytics project on the machine. Verified that file is untouched.

## Verification

Three real Training Runs on 20 migrated Examples, with the default relative `LB_DATA_DIR`:

```
run 1  runs recorded: 1   active-20260915T035348Z mAP50-95=0.4272 epochs=2 imgsz=1024
                          archived: ['weights/best.pt', 'weights/last.pt']
run 2  runs recorded: 2   active-20260915T035426Z mAP50-95=0.4286 epochs=2 imgsz=1024
run 3  runs recorded: 2   (LB_TRACKING=false — nothing recorded)

artifact root: file:///.../lb-track-check2/data/mlflow/artifacts
stray ./mlruns created? False
ultralytics mlflow flag still True
archive size: 77.2 MB for two runs
```

That last line confirms the retention estimate: ~39 MB per run, so a couple of GB after fifty. The README says so rather than leaving it to be discovered.

206 passed / 1 skipped, mypy clean, ruff clean, 4 import contracts kept.

#13 (the `ExperimentTracker` port, for the facts only the service knows — chiefly the Training Set size, which is the x-axis that makes these mAP numbers mean something) is unblocked by this.
